test/integration/helpers/integration-test-editor.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import {
+	BlockEditorKeyboardShortcuts,
+	BlockEditorProvider,
+	BlockList,
+	BlockTools,
+	BlockInspector,
+	WritingFlow,
+	ObserveTyping,
+} from '@wordpress/block-editor';
+import { Popover, SlotFillProvider } from '@wordpress/components';
+import { registerCoreBlocks } from '@wordpress/block-library';
+import { ShortcutProvider } from '@wordpress/keyboard-shortcuts';
+import '@wordpress/format-library';
+import {
+	createBlock,
+	unregisterBlockType,
+	getBlockTypes,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { waitForStoreResolvers } from './wait-for-store-resolvers';
+
+// Polyfill for String.prototype.replaceAll until CI is runnig Node 15 or higher.
+if ( ! String.prototype.replaceAll ) {
+	String.prototype.replaceAll = function ( str, newStr ) {
+		// If a regex pattern
+		if (
+			Object.prototype.toString.call( str ).toLowerCase() ===
+			'[object regexp]'
+		) {
+			return this.replace( str, newStr );
+		}
+
+		// If a string
+		return this.replace( new RegExp( str, 'g' ), newStr );
+	};
+}
+
+/**
+ * For registering all the core blocks if needed as part of test setup.
+ */
+export function registerAllCoreBlocks() {
+	registerCoreBlocks();
+}
+
+/**
+ * Unregisters all blocks if needed as part of test cleanup.
+ */
+export function unRegisterAllBlocks() {
+	getBlockTypes().forEach( ( { name } ) => unregisterBlockType( name ) );
+}
+
+/**
+ * Selects the block to be test by the aria-label on block wrapper, eg. "Block: Cover".
+ *
+ * @param {string}                                  name
+ * @param {import('@testing-library/react').Screen} screen
+ */
+export async function selectBlock( name, screen ) {
+	await userEvent.click( screen.getByLabelText( name ) );
+}
+
+/**
+ * Creates the block to be tested.
+ *
+ * @param {string} name        Block name.
+ * @param {Object} attributes  Block attributes.
+ * @param {?Array} innerBlocks Nested blocks.
+ *
+ * @return {Object} Block object.
+ */
+export function createTestBlock( name, attributes = {}, innerBlocks = [] ) {
```

Comment:
I'm not sure if we need this while we can just import from `@wordpress/blocks` in tests?

Reply:
I was initially thinking we would be doing more manipulation of the block creation, but it ended up that nothing additional was needed for these tests. The only advantage with this currently is that it means the tests don't need another dependency, and it means it would be easier to add additional manipulation later if needed - but I don't have a strong view on that, maybe we should just import from `@wordpress/blocks` in the test file as you suggest - what do you think?

Reply:
Yeah, I think just importing the package is good enough. Or else we'll be just re-implementing the helper in each test, which could be less ergonomic for testers. From another point of view, it's an integration test, so naturally we should be able to use whatever package we use for developing the package to test it.

Reply:
Being handled in initialize editor now.